src/utils/v2/currency.ts

```diff
@@ -0,0 +1,46 @@
+import { V1CurrencyOption } from 'models/v1/currencyOption'
+import { CSSProperties } from 'react'
+
+import { V2CurrencyOption } from 'models/v2/currencyOption'
+
+import { V1_CURRENCY_ETH, V1_CURRENCY_USD } from 'constants/v1/currency'
+import { V2_CURRENCY_ETH, V2_CURRENCY_USD } from 'constants/v2/currency'
+
+const currencies: Record<
```

Comment:
seems like this file and /utils/v1/currency.ts are the same. i'm assuming we have separate files since v2 will support additional currencies? are we sure v2 currencies won't be backwards compatible with the current currency.ts file?

Reply:
@peripheralist they are _pretty_ much the same, with the important difference that they use the `V2CurrencyOption`, which is ETH=1, USD=2, as opposed to V1 which is ETH=0, USD=1.

I took the path of least resistance initially, but I think we can probably consolidate some/all of this currency stuff into 1 file and use generics to switch between currency types. Will take a look!

Reply:
With this one, it turns out to be tricky (impossible?) to create good generic functions here because we're relying on the type name to determine the currency, instead of a property _of_ the type.

For example, one way I think we could improve this is by defining each currency type like this:
```ts
type V1CurrencyETH = {
  id: 0,
  version: '1'
  name: 'ETH'
}
```

This way, we could define a generic function, and switch on `currency.version`. I think this might be out of scope for this PR, but certainly something we can potentially improve. Let me know what you think 🙏

Reply:
ohh whoops, missed that the numbers were different haha. in that case the versioning is interesting, but i'm happy to keep them as separate files